core/src/main/java/org/arquillian/smart/testing/configuration/Configuration.java

```diff
@@ -137,6 +152,24 @@ public void setCustomStrategies(String[] customStrategies) {
         return configItems;
     }
 
+    public List<StrategyConfiguration> strategyConfigurations(String[] strategies) {
+        List<StrategyConfiguration> convertedList = new ArrayList<>();
+
+        new JavaSPILoader().all(StrategyConfiguration.class,
```

Comment:
I am rising a question, instead of creating an ew SPI interface, couldn't we create a new method at `TestExecutionPlannerFactory` which returns this object? So for the implementators needs only to create one SPI

Reply:
We can do that but in that case we don't have configuration for each implementation of `TestExecutionPlannerFactory`. I mean we have configuration only for `Affected`. In other cases it will be only null.

but in this way even for custom strategy with configuration option user can implement it with one spi.

Reply:
I agree with you, but what I see is that from the point of view of developer we can always return `null` or we can create an `Empty` implementation. What I see is that in this way for a developer it is just implementing a new method returning empty class or his own implementation. In the other case you need to start dealing with SPI thing. In my opinion the SPI is the strategy, and the strategy itself is the one which provides you all the information required, let's call it configuration parameters, method call to select tests to execute, ....

Reply:
This is very valid point - I think we should make it happen as part of this PR